Sound effects

JoshMooney opened the pull request:
Added a sound effect controller responsible for managing the sound effects in the game and a soundEfx script to be attached to anything you want to make noise. There is a serializible field in the script for adding audio to the entity and a UISoundEfx script for playing audio when buttons are clicked(Not yet implemented)

JoshMooney committed:
Added AudioController class which manages audio clips being played and the background music

JoshMooney committed:
Added PlayAt to play sounds effects

JoshMooney committed:
Fixed Merge errors and ready for PR

AlexMeuer reviewed (commented):
Tooltips need to be either removed or filled out.

Empty `Start()` and `Update()` methods can be removed.

AlexMeuer committed:
Clean up tooltips and unused methods

Remove empty unity methods from scripts.

Replace tooltip annotation placeholder text with helpful text.

Disable warning for no assignment to variable for sound scripts.
